packages/pluggableWidgets/switch-web/src/Switch.editorPreview.tsx

```diff
@@ -0,0 +1,28 @@
+import { createElement, ReactElement } from "react";
+import { parseStyle } from "@mendix/piw-utils-internal";
+
+import { SwitchPreviewProps } from "../typings/SwitchProps";
+import Switch from "./components/Switch";
+
+export function preview(props: SwitchPreviewProps): ReactElement {
+    // TODO: The widget generator is out of sync with Studio Pro design mode. Change PIW preview props typing (class -> className) generation to remove the ts-ignore below
+    // @ts-ignore
+    const { className, style } = props;
+
+    return (
+        <Switch
+            id="switch-preview"
+            validation={undefined}
+            // Studio Pro will pass `readOnly` value but this is excluded from preview prop typings (via generator)
+            // @ts-ignore
+            editable={!props.readOnly ?? true}
+            isChecked
+            class={className}
+            style={parseStyle(style)}
+        />
+    );
+}
+
+export function getPreviewCss() {
+    return require("./ui/switch-preview.scss");
```

Comment:
Can we reuse switch-main here? The styles are exactly the same.

Reply:
Yes, changed, thanks.

Reply:
@Keraito if you can please review for Isa

Reply:
addressed in https://github.com/mendix/widgets-resources/commit/63e36cb003bfec274e216591ff80160f1538a13f